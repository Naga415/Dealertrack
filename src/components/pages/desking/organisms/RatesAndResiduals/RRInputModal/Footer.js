'use strict';
import React from 'react';
import PropTypes from 'prop-types';
import Button from '@coxautokc/fusion-ui-components/lib/Button';

// eslint-disable-next-line react/prefer-stateless-function
class Footer extends React.PureComponent {
  constructor(props, context) {
    super(props, context);
  }

  render() {
    return (
      <div>
        <Button htmlId="RRModelCancel" onClick={this.props.onCancel}>Cancel</Button>
        <Button htmlId="RRModelApply" onClick={this.props.onApply} bsStyle="primary" disabled={!this.props.applyEnabled}>Apply</Button>
      </div>
    );
  }
}

Footer.propTypes = {
  onCancel: PropTypes.func,
  onApply: PropTypes.func,
  applyEnabled: PropTypes.bool
};

export default Footer;
