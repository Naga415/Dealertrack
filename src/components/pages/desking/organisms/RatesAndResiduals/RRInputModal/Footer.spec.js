/**
 * Created by RBANDERSON on 10/4/2017.
 */
import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import sinon from 'sinon';
import Footer from './Footer';


const onCancel = sinon.spy();
const onApply = sinon.spy();

function setupProps(applyEnabled){
  return {
    applyEnabled,
    onCancel,
    onApply
  };
}

function setupmount (applyEnabled) {
  const props = setupProps(applyEnabled);
  return mount(<Footer {...props} />);
}

describe('Footer', ()=>{
  it('renders <Footer/> ', () => {
    const props = setupProps(true);
    const component = renderer.create(<Footer {...props}/>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render <Footer/> with apply button enabled', () => {
    const component = setupmount(true);
    expect(component.find({id:'RRModelApply',disabled:false}).length).toBe(1);
  });

  it('should render <Footer/> with apply button disabled', () => {
    const component = setupmount(false);
    expect(component.find({id:'RRModelApply',disabled:true}).length).toBe(1);
  });

  it('triggers onApply event when Apply button is clicked', () => {
    const component = setupmount(true);
    component.find(`#RRModelApply`).simulate('click');
    expect(onApply.calledOnce).toEqual(true);
  });
  
  it('triggers onCancel event when Cancel button is clicked', () => {
    const component = setupmount(true);
    component.find(`#RRModelCancel`).simulate('click');
    expect(onApply.calledOnce).toEqual(true);
  });
});
