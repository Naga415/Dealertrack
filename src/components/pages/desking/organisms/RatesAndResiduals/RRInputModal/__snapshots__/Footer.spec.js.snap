// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders <Footer/>  1`] = `
<div>
  <button
    className="btn btn-default"
    disabled={false}
    href={undefined}
    id="RRModelCancel"
    onClick={[Function]}
    type="button"
  >
    Cancel
  </button>
  <button
    className="btn btn-primary"
    disabled={false}
    href={undefined}
    id="RRModelApply"
    onClick={[Function]}
    type="button"
  >
    Apply
  </button>
</div>
`;
